src/logic/wallets/store/middlewares/providerWatcher.js

```diff
@@ -25,6 +28,11 @@ const providerWatcherMware = (store: Store<GlobalState>) => (next: Function) =>
   if (watchedActions.includes(action.type)) {
     switch (action.type) {
       case ADD_PROVIDER: {
+        get3Box(true)
```

Comment:
I think you can use async await here

Reply:
the `await` in this middleware does a mess with the execution of the actions in it and triggers multiple notifications for 'Wallet connected'.

As it's not required to be finished, I moved it to an async function and it's not waiting for it to finish.

Lead to a cleaner code tho.